schemast/mutate.go

```diff
@@ -42,6 +49,7 @@ func Mutate(ctx *Context, mutations ...Mutator) error {
 // the type's Fields and Edges methods to return the desired fields and edges.
 type UpsertSchema struct {
 	Name        string
+	BaseConfig  *BaseSchemaConfig
```

Comment:
"base schema" is not an official name we used in Ent (it's not written anywhere in the docs), but we still use it to describe the schema we embed as the "base" (usually, the zero one, `ent.Schema`), so I can understand why you used it. 

I think we can extend this option by letting users configure the [ast.TypeSpec](https://pkg.go.dev/go/ast#TypeSpec) definition instead using the mutators, and making it more generic. cc @rotemtam, what do you think?

Reply:
I think a nice way to add this is to add a `Mixin` field to `UpsertSchema` and use that to populate 

```go
func (User) Mixin() []ent.Mixin {
   // ...
}

```

somehow, WDYT?

Reply:
@rotemtam That was how I saw it as well, but this way was a much simpler way to do the same thing for the time being. I'm willing to take a shot at it though.

Reply:
Here is a brief attempt at mixin https://github.com/ent/contrib/pull/392